includes/User.php

```diff
@@ -2673,7 +2673,31 @@ public function getLocalAttribute($attr, $cityId = null, $sep = "-") {
 	 * @return string
 	 */
 	public function getGlobalAttribute($attribute, $default=null) {
-		return $this->getOptionHelper($attribute, $default);
+		global $wgEnableReadsFromAttributeService;
+
+		$valueFromMW = $this->getOptionHelper($attribute, $default);
+		if (!empty($wgEnableReadsFromAttributeService)) {
+			$this->compareAttributeValueFromService($valueFromMW, $attribute, $default);
+		}
+
+		return $valueFromMW;
+	}
+
+	private function compareAttributeValueFromService($valueFromMW, $attribute, $default) {
+		$valueFromService = $this->userAttributes()->getAttribute($this->getId(), $attribute, $default);
+		if ($valueFromMW !== $valueFromService) {
+			$this->logAttributeMismatch($valueFromMW, $valueFromService, $attribute, $default);
+		}
+	}
+
+	private function logAttributeMismatch($valueFromMW, $valueFromService, $attribute, $default) {
+		$this->error("USER_ATTRIBUTES attribute_mismatch", [
+				"valueFromMW" => $valueFromMW,
+				"valueFromService" => $valueFromService,
+				"attribute" => $attribute,
+				"default" => $default,
+				"userId" => $this->getId()
+			]);
```

Comment:
Originally I thought to do this with no spaces since it's core code and I wanted to be consistent, but it looks like there's code riddled throughout this file which follows our conventions. So, I'll update :)